src/realm/sync/client.cpp

```diff
@@ -814,7 +724,7 @@ void SessionImpl::initiate_integrate_changesets(std::uint_fast64_t downloadable_
     try {
         bool simulate_integration_error = (m_wrapper.m_simulate_integration_error && !changesets.empty());
         if (simulate_integration_error) {
-            throw IntegrationException(ClientError::bad_changeset, "simulated failure");
+            throw IntegrationException(ErrorCodes::BadChangeset, "simulated failure", ProtocolError::bad_changeset);
```

Comment:
I find it a bit confusing that we have two errors. I guess the plan is to refine this later?